buildSrc/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionBwcSetupPlugin.java

```diff
@@ -0,0 +1,263 @@
+        project.getRootProject().getPluginManager().apply(GlobalBuildInfoPlugin.class);
+
+        BuildParams.getBwcVersions().forPreviousUnreleased((BwcVersions.UnreleasedVersionInfo unreleasedVersion) -> {
+            Project bwcProject = project.project(unreleasedVersion.gradleProjectPath);
+            configureBwcProject(project.project(unreleasedVersion.gradleProjectPath), unreleasedVersion);
+        });
+    }
+
+    private void configureBwcProject(Project project, BwcVersions.UnreleasedVersionInfo unreleasedVersionInfo) {
+        BwcSetupExtension bwcSetupExtension = project.getExtensions().create("bwcSetup", BwcSetupExtension.class, project);
+        bwcSetupExtension.setUnreleasedVersionInfo(project.provider(() -> unreleasedVersionInfo));
+        BwcGitExtension gitExtension = project.getPlugins().apply(InternalBwcGitPlugin.class).getGitExtension();
+        Provider<BwcVersions.UnreleasedVersionInfo> unreleasedVersionInfoProvider = bwcSetupExtension.getUnreleasedVersionInfo();
+        Provider<Version> bwcVersion = unreleasedVersionInfoProvider.map(info -> info.version);
+        gitExtension.setBwcVersion(unreleasedVersionInfoProvider.map(info -> info.version));
+        gitExtension.setBwcBranch(unreleasedVersionInfoProvider.map(info -> info.branch));
+        Provider<File> checkoutDir = bwcSetupExtension.getCheckoutDir();
+        gitExtension.setCheckoutDir(bwcSetupExtension.getCheckoutDir());
+
+        project.getPlugins().apply("distribution");
+        // Not published so no need to assemble
+        project.getTasks().named("assemble").configure(t -> t.setEnabled(false));
+
+        TaskProvider<Task> buildBwcTaskProvider = project.getTasks().register("buildBwc");
+        List<ArchiveProject> archiveProjects = resolveArchiveProjects(checkoutDir.get(), bwcVersion.get());
+
+        for (ArchiveProject archiveProject : archiveProjects) {
+            createBuildBwcTask(
+                bwcSetupExtension,
+                project,
+                bwcVersion,
+                archiveProject.name,
+                archiveProject.getProjectPath(),
+                archiveProject.getDistArchiveFile(),
+                buildBwcTaskProvider
+            );
+
+            registerBwcArtifacts(project, archiveProject);
+        }
+
+        // Create build tasks for the JDBC driver used for compatibility testing
+        String jdbcProjectDir = "x-pack/plugin/sql/jdbc";
+
+        File jdbcProjectArtifact = new File(
+            checkoutDir.get(),
+            jdbcProjectDir + "/build/distributions/x-pack-sql-jdbc-" + bwcVersion.get() + "-SNAPSHOT.jar"
+        );
+
+        createBuildBwcTask(bwcSetupExtension, project, bwcVersion, "jdbc", jdbcProjectDir, jdbcProjectArtifact, buildBwcTaskProvider);
+
+        // make sure no dependencies were added to assemble; we want it to be a no-op
+        project.getTasks().named("assemble").configure(t -> t.setDependsOn(Collections.emptyList()));
```

Comment:
Here we zero out the task dependencies but above we actually disable the task. Should we do one or the other?